### PR TITLE
:white_check_mark: Fix get managed policies test

### PR DESCRIPTION
**Pull request by mitchdawson1982**

### CODE REVIEW

File: tests/api/routers/test_role.py [hunk 1]
```diff
@@ -22,13 +22,16 @@ def test_get_role_by_name(create_iam_role, iam_role_name) -> None:
...
+):
     response = client.get(f"/roles/{iam_role_name}/policies/")
     data = response.json()
+    print(data)
 
-    assert len(data) == 1
+    assert len(data) == 2
     assert response.status_code == 200
-    assert data[0]["PolicyName"] == iam_policy_name
+    assert data[0]["PolicyName"] == "inline-policy"
```
Comment (michaeljcollinsuk) [inline tests/api/routers/test_role.py:34]: Could you do an assert on the second policy to check it is `iam_policy_name`?

File: tests/api/routers/test_role.py [hunk 2]
```diff
@@ -22,13 +22,16 @@ def test_get_role_by_name(create_iam_role, iam_role_name) -> None:
     assert data["RoleName"] == iam_role_name
 
 
-def test_get_managed_policies_for_role(iam_role_name, attach_iam_role_policy, iam_policy_name):
+def test_get_policies_for_role(
+    iam_role_name, attach_iam_role_policy, attach_inline_policy, iam_policy_name
+):
     response = client.get(f"/roles/{iam_role_name}/policies/")
     data = response.json()
+    print(data)
```
Comment (michaeljcollinsuk) [inline tests/api/routers/test_role.py:30]: Can be deleted?

File: app/services/aws_roles.py [hunk 3]
```diff
@@ -44,21 +45,68 @@ def oidc_arn(self, domain: str) -> str:
...
-    def create_role(self):
-        trust_policy = self.get_trust_policy_data()
+    def create_role(self, oidc_user_id):
+        trust_policy = self.get_trust_policy_data(oidc_user_id=oidc_user_id)
         return self.client.create_role(
-            RoleName=self.username, AssumeRolePolicyDocument=json.dumps(trust_policy)
+            RoleName=self.rolename, AssumeRolePolicyDocument=json.dumps(trust_policy)
         )
+
+    def get_policies_for_role(self) -> list[dict]:
```
Comment (michaeljcollinsuk) [inline app/services/aws_roles.py:67]: Could consider adding a unit tests for this method - although not a big deal as there is a test for the `get_policies_by_role_name` which ultimately checks the policies returned

**Review by michaeljcollinsuk — COMMENTED** [review]
LGTM some minor comments

File: tests/api/routers/test_role.py (hunk 1 above)
Comment (mitchdawson1982) [reply tests/api/routers/test_role.py:34]: Agreed

File: tests/api/routers/test_role.py (hunk 2 above)
Comment (mitchdawson1982) [reply tests/api/routers/test_role.py:30]: removed

File: app/services/aws_roles.py (hunk 3 above)
Comment (mitchdawson1982) [reply app/services/aws_roles.py:67]: Now included

**Review by michaeljcollinsuk — APPROVED** [review]
LGTM

**Pull request merged**